### ISSUE TITLE
Live diagnostics can cause multiple incorrect NuGet package restore attempts

### ISSUE THREAD

**Issue by brettfo** [dotnet/interactive#631]
In a new cell start slowly typing `#r "nuget: Newtonsoft.Json"`.  When you finally commit the cell, multiple NuGet packages are added to the restore list.  E.g.,

```
Installing package Newtonsoft.Json.....
Installing package Newtons.....
Installing package Newt.....
```

Resubmitting the cell has the same effect.

Workaround: restart the notebook kernel and execute a cell with the text already present.

**Comment by jonsequitur** [comment]
This is probably related to anonymous commands generated during submission splitting.

**Comment by brettfo** [comment]
Fixed by #635.
